flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -1617,7 +1598,16 @@ object TaskManager {
       }
     }
 
-    val preAllocateMemory: Boolean = streamingMode == StreamingMode.BATCH_ONLY
+    val allocationMode = configuration.getString(
+      ConfigConstants.TASK_MANAGER_MEMORY_ALLOCATION_KEY,
+      ConfigConstants.DEFAULT_TASK_MANAGER_MEMORY_ALLOCATION);
+    val preAllocateMemory: Boolean = allocationMode.equals("eager")
```

Comment:
Could we verify whether we only have "eager" or "lazy" and throw an Exception otherwise?